pkgs/development/python-modules/virtualenv/default.nix

```diff
@@ -1,25 +1,53 @@
 { buildPythonPackage
 , fetchPypi
 , lib
-, recursivePthLoader
+, pythonOlder
+, appdirs
+, contextlib2
+, distlib
+, filelock
+, importlib-metadata
+, importlib-resources
+, pathlib2
+, setuptools_scm
+, six
 }:
 
 buildPythonPackage rec {
   pname = "virtualenv";
-  version = "16.7.9";
+  version = "20.0.21";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "0d62c70883c0342d59c11d0ddac0d954d0431321a41ab20851facf2b222598f3";
+    sha256 = "1kxnxxwa25ghlkpyrxa8pi49v87b7ps2gyla7d1h6kbz9sfn45m1";
+
   };
 
-  # Doubt this is needed - FRidh 2017-07-07
-  pythonPath = [ recursivePthLoader ];
+  buildInputs = [
+    setuptools_scm
+  ];
 
-  patches = [ ./virtualenv-change-prefix.patch ];
+  propagatedBuildInputs = [
+    appdirs
+    distlib
+    filelock
+    six
+  ] ++ lib.optionals (pythonOlder "3.3") [
+    contextlib2
+  ] ++ lib.optionals (pythonOlder "3.4" && !lib.stdenv.isWindows) [
+    pathlib2
+  ] ++ lib.optionals (pythonOlder "3.7") [
+    importlib-resources
+  ] ++ lib.optionals (pythonOlder "3.8") [
+    importlib-metadata
+  ];
 
-  # Tarball doesn't contain tests
-  doCheck = false;
+  # TODO: remove this once https://github.com/NixOS/nixpkgs/issues/81441 is solved
+  postCheck = ''
+    for pycache in $(find $out -name __pycache__) ; do
+      rm -fr ''${pycache}
+    done
+  '';
```

Comment:
Doesn’t this already happen automatically via `pythonRemoveBinBytecodeHook`?

Reply:
no, because they are not in `bin`, but e.g. in `/nix/store/6qzmcll5y0naxng0sjx1vw6hmil94dlf-python3.8-virtualenv-20.0.21/lib/python3.8/site-packages/virtualenv/activation/bash/__pycache__`

Reply:
There is likely bytecode in other folders as well, which we do want to keep.

Please remove this `postCheck`. It's a reproducibility issue that we have in all Python packages because of the upstream pip issue. Creating a fix for just a single package doesn't get us anywhere.